core/libp2p/security/plaintext/plaintext.cpp

```diff
@@ -5,69 +5,134 @@
 
 #include "libp2p/security/plaintext/plaintext.hpp"
 
+#include <functional>
+
+#include "libp2p/crypto/protobuf/keys.pb.h"
+#include "libp2p/peer/peer_id.hpp"
 #include "libp2p/security/error.hpp"
 #include "libp2p/security/plaintext/plaintext_connection.hpp"
-#include "libp2p/security/plaintext/plaintext_session.hpp"
+
+#define PLAINTEXT_OUTCOME_TRY(name, res, conn, cb) \
+  auto name = (res);                               \
+  if (name.has_error()) {                          \
+    conn->close();                                 \
+    cb(name.error());                              \
+    return;                                        \
+  }
+
+OUTCOME_CPP_DEFINE_CATEGORY(libp2p::security, Plaintext::Error, e) {
+  using E = libp2p::security::Plaintext::Error;
+  switch (e) {
+    case E::EXCHANGE_SEND_ERROR:
+      return "Error occured while sending Exchange message to the peer";
+    case E::EXCHANGE_RECEIVE_ERROR:
+      return "Error occured while receiving Exchange message to the peer";
+    case E::INVALID_PEER_ID:
+      return "Received peer id doesn't match actual peer id";
+    case E::EMPTY_PEER_ID:
+      return "Peer Id isn't present in the remote peer multiaddr";
+  }
+  return "Unknown error";
+}
 
 namespace libp2p::security {
+
+  Plaintext::Plaintext(
+      std::shared_ptr<plaintext::ExchangeMessageMarshaller> marshaller,
+      std::shared_ptr<peer::IdentityManager> idmgr)
+      : marshaller_(std::move(marshaller)), idmgr_(std::move(idmgr)) {
+    BOOST_ASSERT(marshaller_);
+    BOOST_ASSERT(idmgr_);
+  }
+
   peer::Protocol Plaintext::getProtocolId() const {
     // TODO(akvinikym) 29.05.19: think about creating SecurityProtocolRegister
-    return "/plaintext/1.0.0";
+    return "/plaintext/2.0.0";
```

Comment:
It would be safer if we kept an old implementation intact, adding a new one as an alternative.

Reply:
1.0.0 was never specified, so no reason to keep it